Lib/multiprocessing/pool.py

```diff
@@ -656,13 +682,14 @@ def __exit__(self, exc_type, exc_val, exc_tb):
 
 class ApplyResult(object):
 
-    def __init__(self, cache, callback, error_callback):
+    def __init__(self, pool, callback, error_callback, cache=None):
```

Comment:
Why the separate `cache` argument?

Reply:
This is a leftover from the last PR.

Originally, in the last PR, I made the cache as a keyword argument to decouple the cache from the pool, as this allows a broader set of strategies to have these separated and therefore reducing dependency cycles and lifetime chains. 

In this PR I am targeting the simplest solution, so I am going to remove the cache argument and just get the cache out of the pool. 

In edee524290929b2b23ae3ab611534998a27bfa5b I have reverted to use always the cache from the pool.